modules/fbs/includes/fbs.reservation.inc

```diff
@@ -326,10 +326,20 @@ function fbs_reservation_branch_name($branch_id) {
  *   The default options.
  */
 function fbs_reservation_default_options($account) {
+  // Load profile2 profile.
+  $profile = ding_user_provider_profile($account);
+
+  // Use a metadata wrapper to access the data.
+  $wrapper = entity_metadata_wrapper('profile2', $profile);
+
+  // Get values.
+  $interest_period = fbs_reservation_default_interest_period($profile);
```

Comment:
Change the function to take in a $account and only load the profile inside this function.